### ISSUE TITLE
[v5] Warning too many classes use cases

### ISSUE THREAD

**Issue by rovansteen** [styled-components/styled-components#2648]
Since upgrading to `5.0.0-beta.5` I get the following warning:
```
Over 50 classes were generated for component box__StyledBox. This happens when some of the props you use for styling have many potential values and we need to make a new CSS class for each variant. Over time the stylesheet will grow and slow down your app.
For these particular CSS rules with high dynamicity, consider using the attrs() method together with a style object.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))`width: 100%;`

  <Component />
```

The component referenced here `StyledBox` is my base component that I use to style everything in combination with `styled-system`. But unless I'm missing something this is not a big deal. **Would it be possible to make this warning account for this use case or at least a way to disable it?**

## System:
 - OS: macOS 10.14.5
 - CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
 - Memory: 279.50 MB / 16.00 GB
 - Shell: 5.3 - /bin/zsh
## Binaries:
 - Node: 12.4.0 - /usr/local/bin/node
 - Yarn: 1.15.2 - ~/Code/application/node_modules/.bin/yarn
 - npm: 6.9.0 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - styled-components: ^5.0.0-beta.5 => 5.0.0-beta.5

**Comment by mxstbr** [comment]
@probablyup even 50 is too low... (reference #2622)

Let's either remove that warning or make the threshold much higher. The reason we have that warning is to catch things like people animating using an interpolation, which 50 is not high enough to reliably catch.

**Comment by probablyup** [comment]
Put back to the original value in https://github.com/styled-components/styled-components/releases/tag/v5.0.0-beta.6
